sdk/test-utils/recorder/src/recorder.ts

```diff
@@ -53,19 +58,58 @@ export class Recorder {
   private assetsJson?: string;
   private variables: Record<string, string>;
 
-  constructor(private testContext?: Test | undefined) {
+  constructor(
+    private testContext?:
+      | MochaTest
+      | VitestContext
+      | undefined,
+  ) {
+    if (!this.testContext) {
+      throw new Error(
+        "Unable to determine the recording file path, testContext provided is not defined.",
+      );
+    }
+
     logger.info(`[Recorder#constructor] Creating a recorder instance in ${getTestMode()} mode`);
     if (isRecordMode() || isPlaybackMode()) {
       if (this.testContext) {
-        this.sessionFile = sessionFilePath(this.testContext);
+        let context:
+          | MochaTest
+          | {
+              suiteTitle: string;
+              testTitle: string;
+            };
+        if (this.testContext instanceof MochaTest) {
```

Comment:
not sure if we should trust `instanceof` here since there could be multiple versions of Mocha... can we define a type guard that just looks at the object properties instead?

Reply:
Good point!